superset/assets/javascripts/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -164,29 +173,17 @@ ${this.props.queryEditor.schema}/${input}`;
             />
           </div>
           <div className="m-t-5">
-            {this.props.queryEditor.schema &&
-              <Select
-                name="select-table"
-                ref="selectTable"
-                isLoading={this.state.tableLoading}
-                value={this.state.tableName}
-                placeholder={`Add a table (${this.state.tableOptions.length})`}
-                autosize={false}
-                onChange={this.changeTable.bind(this)}
-                options={this.state.tableOptions}
-              />
-            }
-            {!this.props.queryEditor.schema &&
-              <Select.Async
-                name="async-select-table"
-                ref="selectTable"
-                value={this.state.tableName}
-                placeholder={"Type to search ..."}
-                autosize={false}
-                onChange={this.changeTable.bind(this)}
-                loadOptions={this.getTableNamesBySubStr.bind(this)}
-              />
-            }
+            <Select
+              name="select-table"
+              ref="selectTable"
+              isLoading={this.state.tableLoading}
+              value={this.state.tableName}
+              placeholder={`Add a table (${this.state.tableOptions.length})`}
+              autosize={false}
+              onChange={this.changeTable.bind(this)}
+              filterOptions={filterOptions}
+              options={this.state.tableOptions}
+            />
```

Comment:
i took out the async loading since it seems like this select handles 25000 records easily. i didn't test for hundreds of thousands of tables. i'll add the async functionality back.